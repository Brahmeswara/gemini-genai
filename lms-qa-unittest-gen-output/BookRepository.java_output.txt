```java
package com.example.lws.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.jdbc.Sql;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
public class BookRepositoryTest {

    @Autowired
    private BookRepository bookRepository;

    @Test
    @Sql(scripts = "/insert-data.sql")
    public void shouldFetchBookById() {
        Long id = 1L;
        Book book = bookRepository.findById(id).orElseThrow();
        assertThat(book.getTitle()).isEqualTo("Book 1");
        assertThat(book.getAuthor()).isEqualTo("Author 1");
    }
}
```