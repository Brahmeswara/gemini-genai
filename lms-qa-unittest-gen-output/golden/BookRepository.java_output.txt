```java
package com.example.lws.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class BookRepositoryTest {

    @Autowired
    private BookRepository bookRepository;

    @Test
    void should_find_no_books_if_repository_is_empty() {
        assertThat(bookRepository.findAll()).isEmpty();
    }

    @Test
    void should_store_a_book() {
        Book book = Book.builder()
                .title("The Hitchhiker's Guide to the Galaxy")
                .author("Douglas Adams")
                .build();

        bookRepository.save(book);

        assertThat(bookRepository.findAll()).hasSize(1);
        assertThat(bookRepository.findAll()).contains(book);
    }

    @Test
    void should_delete_all_books() {
        Book book = Book.builder()
                .title("The Hitchhiker's Guide to the Galaxy")
                .author("Douglas Adams")
                .build();

        bookRepository.save(book);

        bookRepository.deleteAll();

        assertThat(bookRepository.findAll()).isEmpty();
    }
}
```